repos/system_upgrade/el7toel8/actors/cupsmigrate/actor.py

```diff
@@ -0,0 +1,339 @@
+                          summary='CUPS is not installed or its configuration is up-to-date.',
+                          severity='low'
+                        )
+          raise StopActorExecution('CUPS will not be migrated')
+
+        include_files = model.include_files
+        interface = model.migration_matrix.interface
+        digest = model.migration_matrix.digest
+        include = model.migration_matrix.include
+        certkey = model.migration_matrix.certkey
+        env = model.migration_matrix.env
+        printcap = model.migration_matrix.printcap
+
+
+        if interface is True:
+          report_generic(
+                          title='Print queues with interface scripts are used on the machine.',
+                          summary='There are files in /etc/cups/interfaces directory, which indicates usage of print queues with interface scripts. Interface script were removed from CUPS due security issues, so print queues with them will not work in newer CUPS version. Please try to install your printer with printer driver.',
+                          severity='high'
+                        )
+        if include is True:
+          result = self.migrate_includes(include_files)
+          if result is not True:
+            raise StopActorExecutionError('Include directive could not be migrated due IOError')
+
+        if digest is True:
+          result = self.find_and_replace(cupsd_changes_list[DIGEST][NAMES],
+                                         cupsd_changes_list[DIGEST][NEWNAME],
+                                         cupsd_changes_list[DIGEST][SOURCE])
+          if result is not True:
+            raise StopActorExecutionError('Digest directives could not be migrated due IOError')
+
+        if certkey is True:
+          result = self.find_and_replace(cupsfiles_changes_list[CERTKEY][NAMES],
+                                         cupsfiles_changes_list[CERTKEY][NEWNAME],
+                                         cupsfiles_changes_list[CERTKEY][SOURCE])
+          if result is not True:
+            raise StopActorExecutionError('ServerCertificate and ServerKey directives could not be migrated due IOError')
+
+        if env is True:
+          result = self.move(cupsd_changes_list[ENV][NAMES],
+                             cupsd_changes_list[ENV][SOURCE],
+                             cupsd_changes_list[ENV][TARGET])
+          if result is not True:
+            raise StopActorExecutionError('PassEnv/SetEnv directives could not be migrated due IOError')
+
+        if printcap is True:
+          result = self.move(cupsd_changes_list[PRINTCAP][NAMES],
+                             cupsd_changes_list[PRINTCAP][SOURCE],
+                             cupsd_changes_list[PRINTCAP][TARGET])
+          if result is not True:
+            raise StopActorExecutionError('PrintcapFormat directive could not be migrated due IOError')
+
+        report_generic(
```

Comment:
If changes on CUPS are now transparent to user and the system will be working exactly it was before, I would remove this report_generic. If the changes made can have some impact into daily use of system I would add some detailed message about what was changed.

Reply:
No longer creates report in this actor.